### PR TITLE
feat: add inpage provider for webview

### PR DESCRIPTION
**Pull request by Alex-A4**
- [x] Add `js/` folder with inpage-provider connection code
- [x] Add `assets/js`folder with compiled inpage-provider that could be connected to the app
- [x] add/update melos scripts

### CODE REVIEW

File: packages/flutter_nekoton_bridge/assets/js/.gitignore
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/assets/js/.gitignore:1]: Maybe we should use project's .gitignore for this purpose. It's not better for understanding, but keeps clean scripts simple :)

File: scripts/clean.sh [hunk 1]
```diff
@@ -26,6 +26,10 @@ rm -rdf packages/flutter_nekoton_bridge/ios/Frameworks/nekoton_bridge*.zip
 rm -rdf packages/flutter_nekoton_bridge/macos/Frameworks/NekotonBridge.xcframework
 rm -rdf packages/flutter_nekoton_bridge/macos/Frameworks/nekoton_bridge*.zip
 
+# Remove js-related files
+rm -rdf packages/flutter_nekoton_bridge/js/node_modules/
+find packages/flutter_nekoton_bridge/assets/js/ -type f -not -name '.gitignore' -delete
```
Comment (nesquikm) [inline scripts/clean.sh:31]: I suggest to use project's .gitignore so we can just rm this whole folder.

File: packages/flutter_nekoton_bridge/js/.prettierrc
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/.prettierrc:1]: Who calls prettier?

File: packages/flutter_nekoton_bridge/js/.gitignore
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/.gitignore:1]: maybe +1 package-level .gitignore?

File: packages/flutter_nekoton_bridge/js/package.json
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/package.json:1]: Should we configure tests and dependabot for the package?

File: packages/flutter_nekoton_bridge/js/src/errors.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export enum RpcErrorCode {
+    INTERNAL,
```
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/src/errors.ts:2]: Why screaming case? :)

File: packages/flutter_nekoton_bridge/js/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+;(window as Record<string, any>).__hasEverscaleProvider = true
```
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/src/index.ts:1]: wtf is ';'? :)

File: packages/flutter_nekoton_bridge/js/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+;(window as Record<string, any>).hasTonProvider = true
+
+import { Buffer } from 'buffer';
+window.Buffer = Buffer;
+
+// @ts-ignore
+let __define: Define | undefined
+
+const cleanContextForImports = () => {
+    // @ts-ignore
```
Comment (nesquikm) [inline packages/flutter_nekoton_bridge/js/src/index.ts:13]: BUT we have "tabWidth: 2" in .prettierrc! Seems like .prettierrc is only for decorations.

**Review by nesquikm — CHANGES_REQUESTED** [review]
I thing we should call prettier in 'check-format' melos rule. Or in pre-commit hook (but imho melos will be better because we already have github workflow that calls it's 'check-format').

File: packages/flutter_nekoton_bridge/assets/js/.gitignore
Comment (Alex-A4) [reply packages/flutter_nekoton_bridge/assets/js/.gitignore:1]: -

File: scripts/clean.sh (hunk 1 above)
Comment (Alex-A4) [reply scripts/clean.sh:31]: -

File: packages/flutter_nekoton_bridge/js/.prettierrc
Comment (Alex-A4) [reply packages/flutter_nekoton_bridge/js/.prettierrc:1]: IDK, all js-related files were copied from old project

Comment (Alex-A4) [reply packages/flutter_nekoton_bridge/js/.prettierrc:1]: Same for all bottom comments

**Review by nesquikm — APPROVED** [review]
It's okay for now, but we should clean JS-related thins (prettier, case etc) in future.

**Pull request merged**